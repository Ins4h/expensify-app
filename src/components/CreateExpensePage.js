import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";

export class CreateExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

// const CreateExpensePage = ({ dispatch, history, onSubmit }) => (
//   <div>
//     <h1>This is from createExpense component</h1>
//     <ExpenseForm
//       onSubmit={(expense) => {
//         // dispatch(addExpense(expense));
//         onSubmit(expense);
//         history.push("/");
//       }}
//     />
//   </div>
// );

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(CreateExpensePage);
