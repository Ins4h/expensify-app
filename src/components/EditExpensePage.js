import React from "react";
import { connect } from "react-redux";
import { editExpense, removeExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";
import { Link } from "react-router-dom";

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  onClick = (expense) => {
    this.props.removeExpense(this.props.expense.id);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <Link to="/">
          <button onClick={this.onClick}>Remove</button>
        </Link>
      </div>
    );
  }
}

// const EditExpensePage = (props) => (
//   <div>
//     <ExpenseForm
//       expense={props.expense}
//       onSubmit={(expense) => {
//         props.dispatch(editExpense(props.expense.id, expense));
//         props.history.push("/");
//       }}
//     />
//     <Link to="/">
//       <button
//         onClick={(e) => {
//           props.dispatch(removeExpense(props.expense.id));
//         }}
//       >
//         Remove
//       </button>
//     </Link>
//   </div>
// );

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
  };
};

const mapDispatchToProps = (dispatch) => ({
  removeExpense: (id) => dispatch(removeExpense(id)),
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
