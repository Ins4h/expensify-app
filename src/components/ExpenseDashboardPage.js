import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFliter";

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
