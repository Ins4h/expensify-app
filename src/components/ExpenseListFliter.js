import React from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate,
} from "../actions/filters";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    if (e.target.value === "date") this.props.sortByDate();
    else this.props.sortByAmount();
  };
  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              placeholder="Search Expenses"
              className="text-inputs"
              type="text"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select className="select" onChange={this.onSortChange}>
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              startDateId={"start"}
              endDate={this.props.filters.endDate}
              endDateId={"end"}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
});

const mapStateToProps = (state) => ({ filters: state.filters });

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
