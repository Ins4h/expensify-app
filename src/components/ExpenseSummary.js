import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import { Link } from "react-router-dom";
import getExpenseTotal from "../selectors/selectExpenseTotal";
import getVisibleExpenses from "../selectors/expenses";

export const ExpenseSummary = ({ expenses }) => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        Viewing
        <span>
          {" "}
          {expenses.length} {expenses.length === 1 ? " expense" : " expenses"}
        </span>{" "}
        totalling
        <span>{numeral(getExpenseTotal(expenses)).format(" $0.00")}</span>
      </h1>
      <div className="page-header__actions">
        <Link className="button" to="/create">
          Add Expense
        </Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: getVisibleExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseSummary);
