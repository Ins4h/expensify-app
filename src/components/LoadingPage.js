import React from "react";

const LoadingPage = () => (
  <div>
    <img className="loader__image" src="/images/loader.gif" alt="loading" />
  </div>
);

export default LoadingPage;
