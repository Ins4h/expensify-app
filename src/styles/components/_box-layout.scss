.box-layout {
  background-image: url("/images/bg.jpg");
  filter: blur(1.3px);
  background-position: center;
  background-size: cover;
  height: 100vh;
  width: 100vw;

  &__box {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: 2;
    background: fade-out($color: white, $amount: 0.12);
    border-radius: 5px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
  }

  &__title {
    margin: 0;
    line-height: 1;
  }
}
