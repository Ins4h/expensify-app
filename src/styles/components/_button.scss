.button {
  display: inline-block;
  margin: 10px 0 0 0;
  padding: 12px;
  text-decoration: none;
  line-height: 1;
  background: #2471ff;
  border: none;
  border-radius: 2px;
  color: white;
  font-size: 20px;

  &--remove {
    margin: 0;
    background: #888;
  }

  &--logout {
    margin: 0;
    background: none;
  }
}
