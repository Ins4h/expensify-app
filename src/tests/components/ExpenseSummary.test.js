import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import expenses from "../fixtures/expenses";
import { altFilters } from "../fixtures/filters";

test("should render ExpenseSummary with 1 expense correctly", () => {
  const wrapper = shallow(<ExpenseSummary expenses={[expenses[1]]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});
