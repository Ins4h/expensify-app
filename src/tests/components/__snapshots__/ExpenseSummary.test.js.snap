// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
        1
         
         expense
      </span>
       
      totalling
      <span>
         $1500.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing
      <span>
         
        3
         
         expenses
      </span>
       
      totalling
      <span>
         $8500.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
